src/jemalloc.c

```diff
@@ -3315,13 +3308,13 @@ irallocx_prof_sample(tsdn_t *tsdn, void *old_ptr, size_t old_usize,
 
 	alignment = prof_sample_align(alignment);
 	if (usize <= SC_SMALL_MAXCLASS) {
-		p = iralloct(tsdn, old_ptr, old_usize,
-		    SC_LARGE_MINCLASS, alignment, zero, tcache,
-		    arena, hook_args);
+		size_t bumped_usize = PAGE_CEILING(usize);
+		p = arena_prof_ralloc(tsdn, arena, old_ptr, old_usize, bumped_usize,
```

Comment:
Similarly, bumped_usize should be computed with sa2u. With that change, which part of the `iralloct` isn't working?

Reply:
Similar to the reason for not going through `imalloc_no_sample`, if we call `iralloct` the allocation is treated as small.